### ISSUE TITLE
Change the activation function in conv.py in YOLOv8

### ISSUE THREAD

**Issue by Cheryl33990** [ultralytics/ultralytics#7352]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Train

### Bug

In my project, I want to test out the performance of YOLOv8 with different activation function.
I want to **change SiLU to Meta-ACON** , so I add the class in conv.py

```python
class MetaAconC(nn.Module):
    r""" ACON activation (activate or not).
    MetaAconC: (p1*x-p2*x) * sigmoid(beta*(p1*x-p2*x)) + p2*x, beta is generated by a small network
    according to "Activate or Not: Learning Customized Activation" <https://arxiv.org/pdf/2009.04759.pdf>.
    """

    def __init__(self, c1, k=1, s=1, r=16):  # ch_in, kernel, stride, r
        super().__init__()
        # To reduce parameters, add r to rescaled, default=16
        c2 = max(r, c1 // r)
        self.p1 = nn.Parameter(torch.randn(1, c1, 1, 1))
        self.p2 = nn.Parameter(torch.randn(1, c1, 1, 1))
        self.fc1 = nn.Conv2d(c1, c2, k, s, bias=True)
        self.fc2 = nn.Conv2d(c2, c1, k, s, bias=True)
        # self.bn1 = nn.BatchNorm2d(c2)
        # self.bn2 = nn.BatchNorm2d(c1)

    def forward(self, x):
        # Adaptive function include layer-wise, channel-wise, pixel-wise
        # Here is Channel-wise, find the mean of H & W
        # Use sigmoid to get beta
        y = x.mean(dim=2, keepdims=True).mean(dim=3, keepdims=True)
        # batch-size 1 bug/instabilities https://github.com/ultralytics/yolov5/issues/2891
        # beta = torch.sigmoid(self.bn2(self.fc2(self.bn1(self.fc1(y)))))  # bug/unstable
        beta = torch.sigmoid(self.fc2(self.fc1(y)))  # bug patch BN layers removed
        dpx = (self.p1 - self.p2) * x
        return dpx * torch.sigmoid(beta * dpx) + self.p2 * x
```

Second, Change the conv:

```python
class Conv(nn.Module):
    # Standard convolution
    def __init__(self, c1, c2, k=1, s=1, p=None, g=1, act=True):  # ch_in, ch_out, kernel, stride, padding, groups
        super().__init__()
        self.conv = nn.Conv2d(c1, c2, k, s, autopad(k, p), groups=g, bias=False)
        self.bn = nn.BatchNorm2d(c2)
        self.act = MetaAconC(c1=c2) if act is True else (act if isinstance(act, nn.Module) else nn.Identity())

    def forward(self, x):
        return self.act(self.bn(self.conv(x)))

    def forward_fuse(self, x):
        return self.act(self.conv(x))
```

Then, I start to train the model.
However, it shows nan.
![image](https://github.com/ultralytics/ultralytics/assets/87057429/8729baf6-0e05-4912-9a76-e981a7e0bb0f)
The result csv is nan, too.

Could someone please tell me what the problem is?

### Additional

**Reference**
[1]YOLOv5 use Meta-ACON:
https://blog.csdn.net/weixin_43799388/article/details/123603131
[2]Meta-ACON GitHub:
https://github.com/nmaac/acon

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@Cheryl33990 hello! It looks like you've done a good job implementing the Meta-ACON activation function into YOLOv8's `conv.py`. Seeing `nan` values during training can be caused by various issues, such as unstable gradients or parameter initialization problems.

Here are a few steps you can take to troubleshoot the issue:

1. **Initialization**: Ensure that the parameters of Meta-ACON are initialized properly. Sometimes, the default initialization may not be suitable for new activation functions.

2. **Gradient Check**: Perform a gradient check to see if the gradients are exploding or vanishing. This can be done by inspecting the gradients at different layers during the backward pass.

3. **Learning Rate**: Try using a lower learning rate. A high learning rate can sometimes cause instability, especially with new activation functions.

4. **Batch Normalization**: If you've removed batch normalization layers, consider adding them back as they can help stabilize training.

5. **Simplify**: Start with a simpler version of Meta-ACON or test it on a smaller model to ensure it works as expected before scaling up.

6. **Debugging**: Use debugging tools or add print statements to monitor the outputs of the Meta-ACON layer and check for any unexpected behavior.

If you continue to face issues, consider reaching out to the community on the Ultralytics forums or reviewing the documentation on custom activations for additional guidance. Remember, experimentation is key, and sometimes it takes a bit of tweaking to get new components to work correctly with existing architectures. Good luck! 🚀

**Comment by Cheryl33990** [comment]
@glenn-jocher Thanks a lot!
I would follow your advice and keep trying.

**Comment by glenn-jocher** [comment]
You're welcome, @Cheryl33990! If you have further questions or need assistance as you progress, feel free to reach out. Happy coding and best of luck with your experiments! 😊👍
